sklearn/inspection/_plot/tests/test_plot_partial_dependence.py

```diff
@@ -717,6 +831,66 @@ def test_partial_dependence_overwrite_labels(
             assert legend_text[0].get_text() == label
 
 
+def test_get_feature_index():
+    """Check that the behaviour of `_get_feature_index`."""
+    # test when features is already an integer index
+    feature = 1
+    for feature_names in (None, ["a", "b", "c"]):
+        assert _get_feature_index(feature, feature_names) == feature
+
+    # test when features is a string and no feature_names are provided
+    feature = "a"
+    feature_names = None
+    err_msg = "When the feature is a string, `feature_names` should be a list"
+    with pytest.raises(ValueError, match=err_msg):
+        _get_feature_index(feature, feature_names)
+
+    # test when features_names is provided
+    feature = "b"
+    feature_names = ["a", "b", "c"]
+    assert _get_feature_index(feature, feature_names) == 1
+
+    feature = "x"
+    err_msg = "Feature x not in feature_names"
+    with pytest.raises(ValueError, match=err_msg):
+        _get_feature_index(feature, feature_names)
```

Comment:
I think this is partially redundant with the tests in `sklearn/inspection/tests/test_pd_utils.py`. I would rather have all the non redundant part of this test for `_get_feature_index` moved to `test_pd_utils.py` to ease maintenance.